yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,583 @@
+            'duration': 213.0,
+            'availability': 'public',
+            'is_live': False,
+            'was_live': False,
+            'live_status': 'not_live'
+        }
+    }]
+
+    def _real_extract(self, url_from_user):
+        video_id = self._match_id(url_from_user)
+        downloaded_json = self._download_json(url_or_request=self._META_DATA_BASE_URL + video_id, video_id=video_id, note='Downloading video metadata', fatal=False)
+
+        def videoAvailability(y, dic):
+            video_availability = dic['premiumPlan']
+            # 0 - public
+            # 1 - premium only
+
+            if video_availability not in (0, 1):
+                y.report_warning(
+                    'unknown availability code: ' + str(video_availability) + '. The extractor should be fixed')
+                return None
+
+            return y._availability(
+                is_private=False,
+                needs_premium=(video_availability == 1),
+                needs_subscription=False,
+                needs_auth=False,
+                is_unlisted=False)
+
+        content_subdict = try_get(downloaded_json, lambda x: x['content'])
+        video_formats_url = try_get(content_subdict, lambda x: url_or_none(x['contentUrl']))
+        availability = try_get(self, lambda x: videoAvailability(x, downloaded_json))
+
+        if video_formats_url:
+            # Prior to adopting M3U, Rokfin stored videos directly in mp4 files:
+            if video_formats_url.endswith('.mp4'):
+                return self.url_result(url=video_formats_url, video_id=video_id, original_url=url_from_user)
+
+            if not(video_formats_url.endswith('.m3u8')):
+                self.raise_no_formats(msg=f'unsupported video URL {video_formats_url}', expected=False)
+
+            frmts = self._extract_m3u8_formats(m3u8_url=video_formats_url, video_id=video_id, fatal=False)
+            self._sort_formats(frmts)
+        else:
+            frmts = None
+
+            if availability == 'premium_only':
+                # The video is premium only.
+                self.raise_no_formats(msg='premium content', expected=True)
+            else:
+                # We don't know why there is no (valid) video URL present.
+                self.raise_no_formats(msg='unable to proceed due to missing meta data', expected=True)
+
+        downloaded_json = try_get(downloaded_json, lambda x: x.get)
+        created_by = try_get(downloaded_json, lambda x: x('createdBy')).get
+        upload_date_time = try_get(downloaded_json, lambda x: x('creationDateTime'))
+        channel_name = try_get(created_by, lambda x: x('name'))
+        content_subdict = try_get(content_subdict, lambda x: x.get)
+
+        return {
+            'id': video_id,
+            'url': video_formats_url,
+            'original_url': url_from_user,
+            'title': try_get(content_subdict, lambda x: x('contentTitle')),
+            'webpage_url': self._RECOMMENDED_VIDEO_BASE_URL + video_id,
```

Comment:
any reason for this?

Reply:
Can you clarify?

Reply:
Dont set `webpage_url` Unless there is a reason to prefer this URL over the one user provided

Reply:
Apparently, the final part of the user URL exists solely for human consumption and is ignored. To illustrate,

https://www.rokfin.com/stream/10543/Its-A-Crazy-Mess-Regional-Director-Blows-Whistle-On-Pfizers-Vaccine-Trial-Data

points to the same video as 

https://www.rokfin.com/stream/10543

which, again, would point to the exact same video as

https://www.rokfin.com/stream/10543/Hello-World-This-Is-Me-Blah-Blah-Blah

That's also why the const variable _RECOMMENDED_VIDEO_BASE_URL has "recommended" in its name.  I'm adding a relevant commentary to the file.